sql/json-functions-generated-column.md

```diff
@@ -0,0 +1,117 @@
+```
+
+Put a legal JSON string at the location corresponding to the JSON field, and then you can insert JSON into the table. TiDB will parse the text and saves it in a more compact and easy-to-access form.
+
+You can also convert other types of data into JSON using CAST:
+
+```sql
+INSERT INTO person (name, address_info) VALUES ("John", CAST('{"city": "Beijing"}' AS JSON));
+INSERT INTO person (name, address_info) VALUES ("John", CAST('123' AS JSON));
+INSERT INTO person (name, address_info) VALUES ("John", CAST(123 AS JSON));
+```
+
+Now, if you want to query all the users living in Beijing, what should you do? Is it necessary to pull back all the data and then filter in the business layer? Not really. The same with traditional document databases like MongoDB, TiDB supports complex combinations of query conditions in the user side.
+
+```sql
+SELECT id, name FROM person WHERE JSON_EXTRACT(address_info, '$.city') = 'Beijing');
+```
+
+`JSON_EXTRACT` is a function that TiDB supports. If you are familiar with JSON in MySQL 5.7, you must know it. The `JSON_EXTRACT` function in TiDB behaves the same as in MySQL. The meaning of the above function is to extract the `city` field from the `address_info` document. The second argument is a "path expression" and is used to specify which field to extract. The complete syntax description of the path expression is complex. Here are a few simple examples to help you understand its general use:
+
+```sql
+SET @person = '{"name":"John","friends":[{"name":"Forest","age":16},{"name":"Zhang San","gender":"male"}]}';
+
+SELECT JSON_EXTRACT(@person,  '$.name'); -- gets "John"
+SELECT JSON_EXTRACT(@person,  '$.friends[0].age'); -- gets 16
+SELECT JSON_EXTRACT(@person,  '$.friends[1].gender'); -- gets "male"
+SELECT JSON_EXTRACT(@person,  '$.friends[2].name'); -- gets NULL
+``` 
+
+In addition to inserting and querying data, TiDB also support editing JSON. In general, TiDB currently supports the following JSON functions in MySQL 5.7:
+
+- [JSON_EXTRACT](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-extract)
+- [JSON_ARRAY](https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-array)
+- [JSON_OBJECT](https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-object)
+- [JSON_SET](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-set)
+- [JSON_REPLACE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-replace)
+- [JSON_INSERT](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-insert)
+- [JSON_REMOVE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-remove)
+- [JSON_TYPE](https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-type)
+- [JSON_UNQUOTE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-unquote)
+
```

Comment:
If you want to query the specific use, see the [JSON Functions document of MySQL 5.7](https://dev.mysql.com/doc/refman/5.7/en/json-functions.html). If you are a user of MySQL 5.7, you can migrate seamlessly.
->
For more information, see the [JSON Functions document of MySQL 5.7](https://dev.mysql.com/doc/refman/5.7/en/json-functions.html). If you are a user of MySQL 5.7, you can migrate to TiDB seamlessly.